test/functional/services/data_views.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { FtrService } from '../ftr_provider_context';
+
+interface DataViewOptions {
+  name: string;
+  adHoc?: boolean;
+  hasTimeField?: boolean;
+  changeTimestampField?: string;
+}
+
+export class DataViewsService extends FtrService {
+  private readonly retry = this.ctx.getService('retry');
+  private readonly testSubjects = this.ctx.getService('testSubjects');
+  private readonly find = this.ctx.getService('find');
+  private readonly comboBox = this.ctx.getService('comboBox');
+  private readonly header = this.ctx.getPageObjects(['header']).header;
+
+  private async create({
+    name, // Data View title, * will be added automatically
+    adHoc = false, // pass 'true' to have temporary Data View
+    hasTimeField = false, // pass 'true' if Data View has timestamp field
+    changeTimestampField, // optionally override default timestamp field
+  }: DataViewOptions) {
+    await this.testSubjects.existOrFail('indexPatternEditorFlyout');
+    await this.testSubjects.setValue('createIndexPatternTitleInput', name, {
+      clearWithKeyboard: true,
+      typeCharByChar: true,
+    });
+    if (hasTimeField) {
+      await this.retry.waitFor('timestamp field loaded', async () => {
+        const timestampField = await this.testSubjects.find('timestampField');
+        return !(await timestampField.elementHasClass('euiComboBox-isDisabled'));
+      });
+
+      if (changeTimestampField) {
+        await this.comboBox.set('timestampField', changeTimestampField);
+      }
+    }
+    await this.testSubjects.click(adHoc ? 'exploreIndexPatternButton' : 'saveIndexPatternButton');
```

Comment:
Perhaps I'm being dogmatic, but is this click verified somewhere?
[This](https://github.com/elastic/kibana/pull/179321/files/5c53cbc109956c32b3ce884e5ea647db481be5e4#r1549353456) is an example of the feedback I'm getting, regarding verifying clicks.

Reply:
yes, in the tests we verify not only particular the click, but that Data View in switcher is correct:

```
await dataViews.waitForSwitcherToBe('logst*');
```